plugin/interactive.go

```diff
@@ -56,27 +63,39 @@ func Prompt(msg string) (string, error) {
 	inFd := int(os.Stdin.Fd())
 	inGrp, err := tcGetpgrp(inFd)
 	if err != nil {
-		return "", fmt.Errorf("error getting process group controlling stdin: %v", err)
+		return fmt.Errorf("error getting process group controlling stdin: %v", err)
 	}
 	curGrp := unix.Getpgrp()
 
-	var v string
-	if inGrp == curGrp {
-		// We control stdin
-		fmt.Fprintf(os.Stderr, "%s: ", msg)
-		_, err = fmt.Scanln(&v)
-	} else {
-		// Need to get control, prompt, then return control.
+	if inGrp != curGrp {
```

Comment:
The problem might be that Ctrl-C is terminating the daemon when it has stdin. So I'm going to try catching SIGTERM and telling it to stop the context (meaning fn needs to be more opinionated about its signature).